client/state/site-settings/exporter/actions.js

```diff
@@ -83,28 +96,32 @@ export function advancedSettingsFail( siteId, error ) {
  * @param  {Number}   siteId  The ID of the site to export
  * @return {Function}         Action thunk
  */
-export function startExport( siteId ) {
-	return ( dispatch ) => {
+export function startExport( siteId, exportAll = true ) {
+	return ( dispatch, getState ) => {
 		if ( ! siteId ) {
 			return;
 		}
 
 		dispatch( {
 			type: EXPORT_START_REQUEST,
-			siteId: siteId
+			siteId,
+			exportAll,
 		} );
 
+		const state = getState();
+		const advancedSettings = exportAll ? null : prepareExportRequest( state, siteId );
+
 		const success =
 			() => dispatch( exportStarted( siteId ) );
 
 		const failure =
 			error => dispatch( exportFailed( siteId, error ) );
 
 		return wpcom.undocumented()
-			.startExport( siteId )
+			.startExport( siteId, advancedSettings )
 			.then( success )
```

Comment:
This might be early in the game for this feature, since you are still working on it and it's still pretty new, but I find that running the output from the API through some `fromAPI` function to normalize the data has saved me some headaches and refactoring. It can handle renaming variables, validating, etc…

```js
const mapApiStatusToExportState = s => get( {
	finished: EXPORT_STATE_FINISHED,
	running: EXPORT_STATE_RUNNING,
}, s, EXPORT_STATE_UNKNOWN );

const fromApi = response => ( {
	attachmentUrl: response.attachment_url,
	status: mapApiStatusToExportState( response.status ),
} );

…

wpcom.undocumented()
	.startExport( … )
	.then( fromApi )
	.then( success )
	.catch( failure );

…

const success = export => {
	switch ( export.status ) {
		case EXPORT_STATE_FINISHED:
			…
		…
	}
}
```